pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -3093,9 +3236,124 @@ func (r *ClickHouseReader) GetMetricResult(ctx context.Context, query string) ([
 		series := model.Series{Labels: attributes, Points: points}
 		seriesList = append(seriesList, &series)
 	}
+	// err = r.db.Exec(ctx, "DROP TEMPORARY TABLE IF EXISTS getSubTreeSpans"+hash)
+	// if err != nil {
+	// 	zap.S().Error("Error in dropping temporary table: ", err)
+	// 	return nil, err
+	// }
+
 	return seriesList, nil
 }
 
+func getSubTreeAlgorithm(payload []model.SearchSpanResponseItem, getSpansSubQueryDBResponses []model.GetSpansSubQueryDBResponse) (map[string]*model.SearchSpanResponseItem, error) {
+
+	var spans []*model.Span
+	for _, spanItem := range payload {
+		var parentID string
+		if len(spanItem.References) > 0 && spanItem.References[0].RefType == "CHILD_OF" {
+			parentID = spanItem.References[0].SpanId
```

Comment:
why isn't `parentSpanID` part of `SearchSpanResponseItem`?

Reply:
Because we are querying `signoz_spans` table where we are not storing `parentSpanID`.

Reply:
ok..maybe we should add `parentID` in `signoz_spans` but let's hold this for sometime

Reply:
Yes, we are using references string since beginning of project, not sure how it's more useful. Jaeger also uses references.